### PR TITLE
Add three new BigQuery snippets, and address minor nits.

### PR DESCRIPTION
**Pull request by shollyman**
New snippets:
bigquery_update_dataset_description
bigquery_update_dataset_expiration
bigquery_delete_dataset

### CODE REVIEW

File: bigquery/snippets/snippet.go [hunk 1]
```diff
@@ -25,6 +26,49 @@ func createDataset(client *bigquery.Client, datasetID string) error {
 	return nil
 }
 
+func updateDatasetDescription(client *bigquery.Client, datasetID string) error {
```
Comment (tbpg) [inline bigquery/snippets/snippet.go]: Looks like `updateDatasetDescription` checks `ds.Metadata`, but `updateDatasetDefaultExpiration` and `deleteEmptyDataset` don't. Is that intentional? I don't quite understand `original.ETag` and when to use it, either.

Comment (shollyman) [reply bigquery/snippets/snippet.go]: It was intentional.  Per offline discussion with tswast, samples will leverage use of the ETag for safe updates.

File: bigquery/snippets/snippet.go [hunk 2]
```diff
@@ -25,6 +26,53 @@ func createDataset(client *bigquery.Client, datasetID string) error {
...
+	ds := client.Dataset(datasetID)
+	original, err := ds.Metadata(ctx)
+	if err != nil {
+		return err
+	}
+	changes := bigquery.DatasetMetadataToUpdate{
+		Description: "Updated Description.",
+	}
+	_, err = ds.Update(ctx, changes, original.ETag)
+	if err != nil {
```
Comment (broady) [inline bigquery/snippets/snippet.go]: prefer compact error checking:

````
if _, err := ...; err != nil {
   return err
}
```

File: bigquery/snippets/snippet.go [hunk 3]
```diff
@@ -306,7 +354,7 @@ func exportToGCS(client *bigquery.Client, datasetID, tableID, gcsURI string) err
 	return nil
 }
 
-func importJsonExplicitSchema(client *bigquery.Client, datasetID, tableID string) error {
+func importJSONExplicitSchema(client *bigquery.Client, datasetID, tableID string) error {
```
Comment (broady) [inline bigquery/snippets/snippet.go]: thank you!

**Comment by broady** [comment]
LGTM. @tbpg to provide final approval / merging

**Comment by tbpg** [comment]
I'll follow up on error check format.

**Pull request merged**